File: src/test/java/ukalus/persistence/PersistenceTest.java
package ukalus.persistence;

import junit.framework.TestCase;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * @author pmgremo
 *  
 */
public class PersistenceTest  {

  @Before
  public void setUp() throws Exception {
    Persistence.INSTANCE.create("test");
  }

  @After
  public void tearDown() throws Exception {
    Persistence.INSTANCE.close();
    Persistence.INSTANCE.delete("test");
  }

  @Test
  public void testOperations() throws Exception {
    String key = "key";
    String value = "value";
    Persistence.INSTANCE.put(key, value);
    assertEquals(value, Persistence.INSTANCE.get(key));
  }

}

File: src/test/java/ukalus/persistence/PersistenceTest.kt
package ukalus.persistence

import junit.framework.TestCase
import org.junit.After
import org.junit.Before
import org.junit.Test

import org.junit.Assert.assertEquals

/**
 * @author pmgremo
 */
class PersistenceTest {

    @Before
    @Throws(Exception::class)
    fun setUp() {
        Persistence.create("test")
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
        Persistence.close()
        Persistence.delete("test")
    }

    @Test
    @Throws(Exception::class)
    fun testOperations() {
        val key = "key"
        val value = "value"
        Persistence.put(key, value)
        assertEquals(value, Persistence[key])
    }

}